### PR TITLE
Serial hotplug

### PR DESCRIPTION
**Pull request by FengYang**
ID : 1157458 1157436

### CODE REVIEW

File: virttest/utils_test/__init__.py [hunk 1]
```diff
@@ -506,7 +506,7 @@ def run_file_transfer(test, params, env):
 
 
 @error.context_aware
-def run_virtio_serial_file_transfer(test, params, env, port_name=None,
+def run_virtio_serial_file_transfer(test, params, env, port_names=None,
                                     sender="guest", md5_check=True):
```
Comment (ypu) [inline virttest/utils_test/__init__.py]: This update request all test cases and functions calling this to be updated.

File: virttest/utils_test/__init__.py [hunk 2]
```diff
@@ -534,21 +534,39 @@ def run_host_cmd(host_cmd, timeout=720):
...
+                try:
+                    md5_guest = re.findall(md5_re, g_output)[0]
+                except Exception:
+                    err = "Fail to get md5, script may fail."
+                    err += " Script output:\n%s" % g_output
+                    raise error.TestError(err)
+        finally:
+            if host_thread:
+                output = ""
+                output = host_thread.join(10)
```
Comment (ypu) [inline virttest/utils_test/__init__.py]: Do you mean += here?

**Comment by FengYang** [comment]
@xutian  Is anyone still working on this pullreq in redhat?

**Comment by xutian** [comment]
@FengYang 

Do you have time to fix know issue in the pullreq, if no  we can take over it. :)

Thanks,
Xu

**Comment by FengYang** [comment]
Hi @xutian
I do not use autotest in my daily work, so do not have time coding for it. : (  You can  freely update all the reviewing patches under my name.  Thanks

I can help review some patches when I have times. :  )

**Comment by xutian** [comment]
@FengYang 
 Ok, we will take over opening PR. thanks your extra spent on virt-test. 

Xu

File: virttest/qemu_vm.py [hunk 3]
```diff
@@ -431,6 +431,25 @@ def add_serial(devices, name, filename):
             cmd += _add_option("chardev", serial_id)
             return cmd
 
+        def add_chardev_device(devices, name, chardev, filename=None,
+                               device_id=None):
+            """
+            Appends chardev device to cmdline.
+            :param name: Name of the chardev device
+            :param filename: Path to chardev filename
+            :param porttype: Type of the port (*serialport, console)
```
Comment (spcui) [inline virttest/qemu_vm.py]: need delete this line

File: virttest/utils_test/__init__.py (hunk 1 above)
Comment (spcui) [inline virttest/utils_test/__init__.py]: Agreed with ypu's comment.

**Comment by xutian** [comment]
close this one, since fyang no enough time slot to update it.  And open new PR#2216 to fix defeat in comment. And thanks @FengYang  works.

**Pull request closed without merging**